src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptionsUpdateHandler.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Reflection.Metadata;
+using System.Text.Json;
+
+[assembly: MetadataUpdateHandler(typeof(JsonSerializerOptionsUpdateHandler))]
+
+namespace System.Text.Json
+{
+    /// <summary>Handler used to clear JsonSerializerOptions reflection cache upon a metadata update.</summary>
+    internal static class JsonSerializerOptionsUpdateHandler
+    {
+        public static void BeforeUpdate(Type? type)
```

Comment:
What is the argument used for?

Reply:
> Is it subscribing to some convention used by hot reloading?

Yes, which is still being worked out.  I'm expecting this is going to change to actually be `Type[]?`.  Basically the hot reload infrastructure will call this with all of the types it knows are being patched.  In the case of JsonSerializer, it's simplest / least impactful on the hot path to just clear everything in the case of an edit, so it's ignoring the type.

Reply:
Is the plan to call this using reflection or will the method be made public pending API approval?

Reply:
Via reflection.  The hot reload agent finds all MetadataUpdateHandler attributes,  finds all of the specified handiers,  and invokes the corresponding methods they implement, all via reflection.